pubsublite-spark-sql-streaming/src/test/java/com/google/cloud/pubsublite/spark/PslSourceOffsetTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.spark;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.pubsublite.*;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+public class PslSourceOffsetTest {
+
+  private static final SubscriptionPath TEST_SUB;
+
+  static {
+    TEST_SUB =
+        SubscriptionPath.newBuilder()
+            .setProject(ProjectId.of("test-project"))
+            .setLocation(CloudZone.of(CloudRegion.of("us-central1"), 'a'))
+            .setName(SubscriptionName.of("test-sub"))
+            .build();
+  }
+
+  @Test
+  public void roundTrip() {
+    PslSourceOffset offset =
+        new PslSourceOffset(
+            ImmutableMap.of(
+                Partition.of(3L), Offset.of(10L),
+                Partition.of(1L), Offset.of(5L),
+                Partition.of(2L), Offset.of(8L)));
+    assertThat(offset.json()).isEqualTo("{\"1\":5,\"2\":8,\"3\":10}");
+    assertThat(PslSourceOffset.fromJson(offset.json())).isEqualTo(offset);
+  }
+
+  @Test
+  public void mergePslSourceOffsets() {
+    PslSourceOffset o1 =
+        new PslSourceOffset(
+            ImmutableMap.of(
+                Partition.of(3L), Offset.of(10L),
```

Comment:
nit, please order these for ease of reading?

Reply:
This is intended to test the serialization will make it sorted. Added comments.

Reply:
There are no comments here?

Reply:
Ah, were you referring to above? You're not testing the serialization in this function, so it doesn't make sense to have it disordered. please fix.

Reply:
Done.